src/components/MoneyRequestReportView/MoneyRequestReportTransactionList.tsx

```diff
@@ -74,13 +111,43 @@ function MoneyRequestReportTransactionList({transactions}: MoneyRequestReportTra
         }));
     }, [sortBy, sortOrder, transactions]);
 
+    const navigateToTransaction = useCallback(
+        (activeTransaction: OnyxTypes.Transaction) => {
+            const iouAction = getIOUActionForTransactionID(reportActions, activeTransaction.transactionID);
+            const reportIDToNavigate = iouAction?.childReportID;
+            if (!reportIDToNavigate) {
+                return;
+            }
+
+            const backTo = Navigation.getActiveRoute();
+
+            // Single transaction report will open in RHP, and we need to find every other report ID for the rest of transactions
+            // to display prev/next arrows in RHP for navigating between transactions
+            const sortedSiblingTransactionReportIDs = getThreadReportIDsForTransactions(reportActions, sortedData.transactions);
```

Comment:
We need to filter out pending delete transactions to avoid [issue](https://github.com/Expensify/App/issues/72877) of navigating to those transactions.